src/windows_msvc.rs

```diff
@@ -183,7 +182,10 @@ fn try_bin_dir_impl(mut root_dir: PathBuf, x86_bin: &str, x64_bin: &str, arch: A
     }
 }
 
-fn try_rc_exe(mut pb: PathBuf) -> Option<PathBuf> {
-    pb.push("rc.exe");
-    if pb.exists() { Some(pb) } else { None }
+fn try_tool(tool: &str) -> Box<dyn Fn(PathBuf) ->  Option<PathBuf>> {
```

Comment:
This looks like it can be `fn try_tool(mut pb: PathBuf, tool: &str) -> Option<PathBuf>` and the call-sites `.and_then(|pb| try_tool(pb, tool))` instead.